### PR TITLE
Exclude APIs required only for tests from main bundle

### PR DESCRIPTION
**Pull request by L2jLiga**
bundle.js before: 781kb, after: 776kb

@fabiosantoscode, @nifgraup could you take a look? Actually I'm not sure is it minor or major change according to semver. If this were part of the public APIs, then it should be part of the next major update otherwise it looks like as minor.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,12 +52,14 @@
...
-    "test": "rimraf dist/* && npm run prepare && istanbul instrument dist/bundle.min.js > dist/bundle.instrumented.js && node test/run-tests.js",
+    "test": "rimraf dist/* && npm run build -- --input=main.tests.js && npm run minify && node test/run-tests.js",
     "coverage": "istanbul cover test/run-tests.js",
     "coveralls": "coveralls < coverage/lcov.info",
     "lint": "eslint lib",
     "lint-fix": "eslint --fix lib",
-    "prepare": "rollup -c && cd dist && cross-env TERSER_NO_BUNDLE=1 ../bin/uglifyjs bundle.js -mc --source-map 'content=bundle.js.map,includeSources=true,url=bundle.min.js.map' -o bundle.min.js",
+    "build": "rollup -c",
+    "minify": "cd dist && cross-env TERSER_NO_BUNDLE=1 ../bin/uglifyjs bundle.js -mc --source-map 'content=bundle.js.map,includeSources=true,url=bundle.min.js.map' -o bundle.min.js",
+    "prepare": "npm run build && npm run minify",
```
Comment (fabiosantoscode) [inline package.json]: Can you leave these changes out? I'm working on making the scripts better, but haven't committed it.

Comment (fabiosantoscode) [reply package.json]: The package.json changes I mean

File: main.js [hunk 2]
```diff
@@ -30,11 +30,3 @@ export {
...
 } from "./lib/propmangle.js";
 import "./lib/mozilla-ast.js";
-
-// TESTS
-export * from "./lib/ast.js";
-export {
-    JS_Parse_Error,
-    tokenizer,
-} from "./lib/parse.js";
-export { to_ascii } from "./lib/minify.js";
```
Comment (fabiosantoscode) [inline main.js]: I think only `to_ascii` deserves to die. JS_Parse_Error can be used to tell what exception is being thrown by terser, inside a catch statement.

File: package.json (hunk 1 above)
Comment (L2jLiga) [reply package.json]: Unfortunately no ☹️
We need somehow detect that we running tests and then build Terser with this APIs.. PR can wait until you push your changes and then I'll fix conflicts

Comment (fabiosantoscode) [reply package.json]: No, I'll handle the conflicts, don't worry!

File: main.js (hunk 2 above)
Comment (L2jLiga) [reply main.js]: I guess Terser can use all of this inside and this imports are not required for anything rather then tests

Comment (L2jLiga) [reply main.js]: We can keep some of them if someone wants to use this in the CLI for example

**Comment by fabiosantoscode** [comment]
Lookin' good. Merging it in!

**Comment by MattiasBuelens** [comment]
Wish I had seen this sooner! I added a `map_from_object` helper in #300, but that's only used by the test runner.

I'll make a pull request to move it to the test-only APIs. 😉

**Pull request merged**